test/hotspot/jtreg/serviceability/jvmti/events/ClassPrepare/classprep01/libclassprep01.cpp

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright (c) 2003, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+#include "jvmti.h"
+
+#include "jvmti_common.h"
+
+
+extern "C" {
+
+
+#define PASSED 0
+#define STATUS_FAILED 2
+
+#define EXP_STATUS (JVMTI_CLASS_STATUS_VERIFIED | JVMTI_CLASS_STATUS_PREPARED)
+
+typedef struct {
+  char *sig;
+  jint status;
+  jint mcount;
+  jint fcount;
+  jint icount;
+} writable_class_info;
+
+typedef struct {
+  const char *sig;
+  jint status;
+  jint mcount;
+  jint fcount;
+  jint icount;
+} class_info;
+
+static jvmtiEnv *jvmti = NULL;
+static jvmtiEventCallbacks callbacks;
+static jint result = PASSED;
+static volatile size_t eventsCount = 0; // TODO these 2 vars mofified from different threads in getReady/check. What to DO???
```

Comment:
I'd suggest to use RawMonitorLocker with event_lock or counter_lock to protect updates of these counters.
You can remove this comment then.

Reply:
The variable eventsCount seems to be used in the sinlge thread only. I looked on the several tests with same variable and similar logic.